File: game/src/main/java/game/character/Character.java
package game.character;

import com.google.gson.annotations.Expose;
import game.layout.CellContent;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

/**
 * @author ismd
 */
@Entity
public class Character extends CellContent {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Expose
    private int id;
    @Expose
    private long idUser;
    @Expose
    private String name;
    @Expose
    private int level;
    @Expose
    private long money;
    @Expose
    private int idLayout;
    @Expose
    private int x;
    @Expose
    private int y;
    @Expose
    private int strength;
    @Expose
    private int dexterity;
    @Expose
    private int endurance;
    @Expose
    private int hp;
    @Expose
    private int maxHp;
    @Expose
    private int minDamage;
    @Expose
    private int maxDamage;
    @Expose
    private String image;
    @Expose
    private long experience;

    public Character() {
    }

    public long getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public Character setName(String name) {
        this.name = name;
        return this;
    }

    public long getIdUser() {
        return idUser;
    }

    public Character setIdUser(long idUser) {
        this.idUser = idUser;
        return this;
    }

    public int getLevel() {
        return level;
    }

    public Character setLevel(int level) {
        this.level = level;
        return this;
    }

    public long getMoney() {
        return money;
    }

    public Character setMoney(long money) {
        this.money = money;
        return this;
    }

    public int getIdLayout() {
        return idLayout;
    }

    public Character setIdLayout(int idLayout) {
        this.idLayout = idLayout;
        return this;
    }

    public int getX() {
        return x;
    }

    public Character setX(int x) {
        this.x = x;
        return this;
    }

    public int getY() {
        return y;
    }

    public Character setY(int y) {
        this.y = y;
        return this;
    }

    public int getStrength() {
        return strength;
    }

    public Character setStrength(int strength) {
        this.strength = strength;
        return this;
    }

    public int getDexterity() {
        return dexterity;
    }

    public Character setDexterity(int dexterity) {
        this.dexterity = dexterity;
        return this;
    }

    public int getEndurance() {
        return endurance;
    }

    public Character setEndurance(int endurance) {
        this.endurance = endurance;
        return this;
    }

    public int getHp() {
        return hp;
    }

    public Character setHp(int hp) {
        this.hp = hp;
        return this;
    }

    public int getMaxHp() {
        return maxHp;
    }

    public Character setMaxHp(int maxHp) {
        this.maxHp = maxHp;
        return this;
    }

    public int getMinDamage() {
        return minDamage;
    }

    public Character setMinDamage(int minDamage) {
        this.minDamage = minDamage;
        return this;
    }

    public int getMaxDamage() {
        return maxDamage;
    }

    public Character setMaxDamage(int maxDamage) {
        this.maxDamage = maxDamage;
        return this;
    }

    public String getImage() {
        return image;
    }

    public Character setImage(String image) {
        this.image = image;
        return this;
    }

    public long getExperience() {
        return experience;
    }

    public Character setExperience(long experience) {
        this.experience = experience;
        return this;
    }
}

File: site/application/models/abstract/Character.php
<?php
/**
 * Абстрактный класс персонажей
 * @author ismd
 */

abstract class Abstract_Character extends PsModel {

    private $_id;     // id в таблице Character (сам персонаж)
    private $_idUser; // id в таблице User (пользователь)
    private $_name;
    private $_level;
    private $_money;
    private $_hp;
    private $_maxHp;
    private $_minDamage;
    private $_maxDamage;
    private $_experience;
    private $_image;
    private $_strength;
    private $_dexterity;
    private $_endurance;
    private $_idLayout;
    private $_x;
    private $_y;

    public function toArray() {
        return array(
            'id'         => $this->getId(),
            'idUser'     => $this->getIdUser(),
            'name'       => $this->getName(),
            'level'      => $this->getLevel(),
            'money'      => $this->getMoney(),
            'hp'         => $this->getHp(),
            'maxHp'      => $this->getMaxHp(),
            'minDamage'  => $this->getMinDamage(),
            'maxDamage'  => $this->getMaxDamage(),
            'experience' => $this->getExperience(),
            'image'      => $this->getImage(),
            'strength'   => $this->getStrength(),
            'dexterity'  => $this->getDexterity(),
            'endurance'  => $this->getEndurance(),
            'idLayout'   => $this->getIdLayout(),
            'x'          => $this->getX(),
            'y'          => $this->getY(),
        );
    }

    public function setId($value) {
        $this->_id = (int)$value;
        return $this;
    }

    public function getId() {
        return $this->_id;
    }

    public function setIdUser($value) {
        $this->_idUser = (int)$value;
        return $this;
    }

    public function getIdUser() {
        return $this->_idUser;
    }

    public function setName($value) {
        $this->_name = (string)$value;
        return $this;
    }

    public function getName() {
        return $this->_name;
    }

    public function setLevel($value) {
        $this->_level = (int)$value;
        return $this;
    }

    public function getLevel() {
        return $this->_level;
    }

    public function setMoney($value) {
        $this->_money = (int)$value;
        return $this;
    }

    public function getMoney() {
        return $this->_money;
    }

    public function setHp($value) {
        $this->_hp = (int)$value;
        return $this;
    }

    public function getHp() {
        return $this->_hp;
    }

    public function setMaxHp($value) {
        $this->_maxHp = (int)$value;
        return $this;
    }

    public function getMaxHp() {
        return $this->_maxHp;
    }

    public function setMinDamage($value) {
        $this->_minDamage = (int)$value;
        return $this;
    }

    public function getMinDamage() {
        return $this->_minDamage;
    }

    public function setMaxDamage($value) {
        $this->_maxDamage = (int)$value;
        return $this;
    }

    public function getMaxDamage() {
        return $this->_maxDamage;
    }

    public function setExperience($value) {
        $this->_experience = (int)$value;
        return $this;
    }

    public function getExperience() {
        return $this->_experience;
    }

    public function setImage($value) {
        $this->_image = (string)$value;
        return $this;
    }

    public function getImage() {
        return $this->_image;
    }

    public function setStrength($value) {
        $this->_strength = (int)$value;
        return $this;
    }

    public function getStrength() {
        return $this->_strength;
    }

    public function setDexterity($value) {
        $this->_dexterity = (int)$value;
        return $this;
    }

    public function getDexterity() {
        return $this->_dexterity;
    }

    public function setEndurance($value) {
        $this->_endurance = (int)$value;
        return $this;
    }

    public function getEndurance() {
        return $this->_endurance;
    }

    public function setIdLayout($value) {
        $this->_idLayout = (int)$value;
        return $this;
    }

    public function getIdLayout() {
        return $this->_idLayout;
    }

    public function setX($value) {
        $this->_x = (int)$value;
        return $this;
    }

    public function getX() {
        return $this->_x;
    }

    public function setY($value) {
        $this->_y = (int)$value;
        return $this;
    }

    public function getY() {
        return $this->_y;
    }
}
